### ISSUE TITLE
Evaluation result on egien test split

### ISSUE THREAD

**Issue by GoYchen** [mrharicot/monodepth#3]
Hi Clement,

I have downloaded your pretrained-model on eigen split. After calling
```
python monodepth_main.py --mode test --data_path KITTI_raw_data/ \
    --filenames_file utils/filenames/eigen_test_files.txt \
    --output_directory UnsuperviseMono/eigen_pretrain \
    --checkpoint_path UnsuperviseMono/pretrain_model/model_eigen
```
I got disparities.npy. Then I called

```
python utils/evaluate_kitti.py --split eigen --predicted_disp_path UnsuperviseMono/eigen_pretrain/disparities.npy \
    --gt_path KITTI_raw_data/
```
to get the result
```
0 files missing
   abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.9966,    15.9183,     19.672,      6.374,      0.000,      0.000,      0.000,      0.000
```
which is not consistent with that of your paper.

**But the result for KITTI 200 training images is consistent with that in your paper.**

I checked the evaluation code in evaluate_kiiti.py, and found that you have commented some of the codes

```
    elif args.split == 'eigen':
        num_samples = 697
        test_files = read_text_lines(args.gt_path + 'eigen_test_files.txt')
        gt_files, gt_calib, im_sizes, im_files, cams = read_file_data(test_files, args.gt_path)

        num_test = len(im_files)
        gt_depths = []
        pred_depths = []
        for t_id in range(num_samples):
            camera_id = cams[t_id]  # 2 is left, 3 is right
            depth = generate_depth_map(gt_calib[t_id], gt_files[t_id], im_sizes[t_id], camera_id, False, True)
            gt_depths.append(depth.astype(np.float32))

            disp_pred = cv2.resize(pred_disparities[t_id], (im_sizes[t_id][1], im_sizes[t_id][0]), interpolation=cv2.INTER_LINEAR)
            # disp_pred = disp_pred * disp_pred.shape[1]

            # # need to convert from disparity to depth
            # focal_length, baseline = get_focal_length_baseline(gt_calib[t_id], camera_id)
            # depth_pred = (baseline * focal_length) / disp_pred
            # depth_pred[np.isinf(depth_pred)] = 0

            pred_depths.append(disp_pred)
```

I try to uncomment it, but the result is still weird. Could you help to have a look at it?

Thank you for your help and time.

**Comment by mrharicot** [comment]
Hi GoYchen,

We do not crop the results on the KITTI split, however we crop the ground truth on the Eigen split in order to match the previous papers.
There are two options: `--garg_crop` and `--eigen_crop`.
We used the `garg_crop` for all results except for Eigen's where we used `eigen_crop` in order to match as the results from their paper.

If you use `--garg_crop` on `evaluate_kitti.py` the output should match the result reported in our paper.
